src/Features/Core/Diagnostics/BaseDiagnosticIncrementalAnalyzer.cs

```diff
@@ -62,6 +62,13 @@ protected BaseDiagnosticIncrementalAnalyzer(DiagnosticAnalyzerService owner, Wor
         /// <returns></returns>
         public abstract Task DocumentOpenAsync(Document document, CancellationToken cancellationToken);
         /// <summary>
+        /// Respond to a document being closed in the host.
+        /// </summary>
+        /// <param name="document">The closed document.</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public abstract Task DocumentCloseAsync(Document document, CancellationToken cancellationToken);
+        /// <summary>
         /// Flush cached diagnostics produced by a prior analysis of a document.
```

Comment:
`DocumentResetAsync` is re-analyzer right? Probably, worth adding that to the xml doc comment here so that the implementer knows the difference between these 2 APIs.